lib/base/process.cpp

```diff
@@ -716,7 +719,11 @@ void Process::IOThreadProc(int tid)
 				bool is_timeout = false;
 
 				if (it->second->m_Timeout != 0) {
-					double timeout = it->second->m_Result.ExecutionStart + it->second->m_Timeout;
+					double timeout = it->second->m_Result.ExecutionStart + it->second->m_Timeout
+#ifndef _WIN32
+						* (it->second->m_SentSigterm ? 1.1 : 1.0)
+#endif /* _WIN32 */
+						;
```

Comment:
I think just doing the following would be nicer, but also see the general comment about adding a new method for this.

```cpp
#ifndef _WIN32
    if (it->second->m_SentSigterm) {
        timeout *= 1.1;
    }
#endif
```